diff --git a/BUILD b/BUILD
index 387dc669d5b58aa38bac4f25a04850ef4a8bd509..3e399bacb5517ad0995c961f89b4f9e16537a8e5 100644
--- a/BUILD
+++ b/BUILD
@@ -2,6 +2,18 @@ package(default_visibility = ["//visibility:public"])
 
 filegroup(name = "empty")
 
+alias(
+    name = "dwp_files",
+    actual = select({
+        ":linux": "@emscripten_bin_linux//:dwp_files",
+        ":linux_arm64": "@emscripten_bin_linux_arm64//:dwp_files",
+        ":macos": "@emscripten_bin_mac//:dwp_files",
+        ":macos_arm64": "@emscripten_bin_mac_arm64//:dwp_files",
+        ":windows": "@emscripten_bin_win//:dwp_files",
+        "//conditions:default": ":empty",
+    }),
+)
+
 platform(
     name = "platform_wasm",
     constraint_values = [
diff --git a/MODULE.bazel b/MODULE.bazel
index e5ea27d66df377651f9e8a6ce75b0cbaaa56c241..4149ef6c886205accd57472fa7eb0818c1405452 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -29,44 +29,6 @@ use_repo(emscripten_deps, "emscripten_bin_mac")
 use_repo(emscripten_deps, "emscripten_bin_mac_arm64")
 use_repo(emscripten_deps, "emscripten_bin_win")
 
-npm = use_extension(
-    "@aspect_rules_js//npm:extensions.bzl",
-    "npm",
-)
-npm.npm_translate_lock(
-    name = "emscripten_npm_linux",
-    data = ["@emscripten_bin_linux//:emscripten/package.json"],
-    npm_package_lock = "@emscripten_bin_linux//:emscripten/package-lock.json",
-)
-npm.npm_translate_lock(
-    name = "emscripten_npm_linux_arm64",
-    data = ["@emscripten_bin_linux_arm64//:emscripten/package.json"],
-    npm_package_lock = "@emscripten_bin_linux_arm64//:emscripten/package-lock.json",
-)
-npm.npm_translate_lock(
-    name = "emscripten_npm_mac",
-    data = ["@emscripten_bin_mac//:emscripten/package.json"],
-    npm_package_lock = "@emscripten_bin_mac//:emscripten/package-lock.json",
-)
-npm.npm_translate_lock(
-    name = "emscripten_npm_mac_arm64",
-    data = ["@emscripten_bin_mac_arm64//:emscripten/package.json"],
-    npm_package_lock = "@emscripten_bin_mac_arm64//:emscripten/package-lock.json",
-)
-npm.npm_translate_lock(
-    name = "emscripten_npm_win",
-    data = ["@emscripten_bin_win//:emscripten/package.json"],
-    npm_package_lock = "@emscripten_bin_win//:emscripten/package-lock.json",
-)
-use_repo(
-    npm,
-    "emscripten_npm_linux",
-    "emscripten_npm_linux_arm64",
-    "emscripten_npm_mac",
-    "emscripten_npm_mac_arm64",
-    "emscripten_npm_win",
-)
-
 emscripten_cache = use_extension("//:emscripten_cache.bzl", "emscripten_cache")
 use_repo(emscripten_cache, "emscripten_cache")
 
diff --git a/emscripten_build_file.bzl b/emscripten_build_file.bzl
index 0d7aff28ca543b3291620283416ae39f161eed74..eefbe79f51ce966204d7e82f6032ff6fcc9a4dc8 100644
--- a/emscripten_build_file.bzl
+++ b/emscripten_build_file.bzl
@@ -90,4 +90,12 @@ filegroup(
         ],
     ),
 )
+
+filegroup(
+    name = "dwp_files",
+    srcs = [
+        "bin/llvm-dwp{bin_extension}",
+    ],
+)
+
 """
diff --git a/emscripten_toolchain/emar.sh b/emscripten_toolchain/emar.sh
index b4ead6ef9b0c6a8d5dee7f0ab58fef391a7dff36..e54e07c66b4b968a2185f388031a653ed506746a 100755
--- a/emscripten_toolchain/emar.sh
+++ b/emscripten_toolchain/emar.sh
@@ -2,4 +2,4 @@
 
 source $(dirname $0)/env.sh
 
-exec python3 $EMSCRIPTEN/emar.py "$@"
+exec "$EM_PYTHON3" $EMSCRIPTEN/emar.py "$@"
diff --git a/emscripten_toolchain/emcc.sh b/emscripten_toolchain/emcc.sh
index 5fdaf9c2958a2f8c1782513806d5697cfe4ba281..e080b4a161b32fc647545d5d78bc0c77e61ea5ce 100755
--- a/emscripten_toolchain/emcc.sh
+++ b/emscripten_toolchain/emcc.sh
@@ -2,4 +2,4 @@
 
 source $(dirname $0)/env.sh
 
-exec python3 $EMSCRIPTEN/emcc.py "$@"
+exec "$EM_PYTHON3" $EMSCRIPTEN/emcc.py "$@"
diff --git a/emscripten_toolchain/emcc_link.sh b/emscripten_toolchain/emcc_link.sh
index 44f32353ba51c3dd1924e7684e6a05c44bd1261a..35586440c61cda3fa0f4682da76ee62dd770f810 100755
--- a/emscripten_toolchain/emcc_link.sh
+++ b/emscripten_toolchain/emcc_link.sh
@@ -2,4 +2,4 @@
 
 source $(dirname $0)/env.sh
 
-exec python3 $(dirname $0)/link_wrapper.py "$@"
+exec "$EM_PYTHON3" $(dirname $0)/link_wrapper.py "$@"
diff --git a/emscripten_toolchain/toolchain.bzl b/emscripten_toolchain/toolchain.bzl
index 80f55f6a49537a2964eb0165c77eba28c54c9234..96a1caa828d07c43c5abda21fbd144e7ac9cceea 100644
--- a/emscripten_toolchain/toolchain.bzl
+++ b/emscripten_toolchain/toolchain.bzl
@@ -15,6 +15,7 @@ load(
     _flag_set = "flag_set",
 )
 load("@bazel_tools//tools/build_defs/cc:action_names.bzl", "ACTION_NAMES")
+load("@rules_python//python:py_runtime_info.bzl", "PyRuntimeInfo")
 
 def flag_set(flags = None, features = None, not_features = None, **kwargs):
     """Extension to flag_set which allows for a "simple" form.
@@ -73,6 +74,9 @@ def _impl(ctx):
     emscripten_dir = ctx.attr.emscripten_binaries.label.workspace_root
 
     nodejs_path = ctx.file.nodejs_bin.path
+    python_runtime = ctx.attr.python_bin[PyRuntimeInfo]
+    python_path = python_runtime.interpreter.path
+    python_home = ctx.attr.python_bin.label.workspace_root
 
     builtin_sysroot = emscripten_dir + "/emscripten/cache/sysroot"
 
@@ -1077,6 +1081,14 @@ def _impl(ctx):
                     key = "NODE_JS_PATH",
                     value = nodejs_path,
                 ),
+                env_entry(
+                    key = "EM_PYTHON3",
+                    value = python_path,
+                ),
+                env_entry(
+                    key = "PYTHONHOME",
+                    value = python_home,
+                ),
             ],
         ),
         # Use llvm backend.  Off by default, enabled via --features=llvm_backend
@@ -1152,6 +1164,7 @@ emscripten_cc_toolchain_config_rule = rule(
         "em_config": attr.label(mandatory = True, allow_single_file = True),
         "emscripten_binaries": attr.label(mandatory = True, cfg = "exec"),
         "nodejs_bin": attr.label(mandatory = True, allow_single_file = True),
+        "python_bin": attr.label(mandatory = True, cfg = "exec"),
         "script_extension": attr.string(mandatory = True, values = ["sh", "bat"]),
     },
     provides = [CcToolchainConfigInfo],
diff --git a/emscripten_toolchain/wasm_cc_binary.bzl b/emscripten_toolchain/wasm_cc_binary.bzl
index e7bd8ef6dc12bde7c0333fd83aacc50d6619208e..fbe86a4c0ab1358e2d0226d5a23f60b74700e448 100644
--- a/emscripten_toolchain/wasm_cc_binary.bzl
+++ b/emscripten_toolchain/wasm_cc_binary.bzl
@@ -151,7 +151,6 @@ def _wasm_cc_binary_legacy_impl(ctx):
     args.add("--allow_empty_outputs")
     args.add_all("--archive", ctx.files.cc_target)
     args.add_joined("--outputs", outputs, join_with = ",")
-
     ctx.actions.run(
         inputs = ctx.files.cc_target,
         outputs = outputs,
diff --git a/remote_emscripten_repository.bzl b/remote_emscripten_repository.bzl
index 819a0480802de37495fee84869c03c0b3b538686..45d2d4b65b6c2ad24ebe25bb9b99d12ecdfa1b29 100644
--- a/remote_emscripten_repository.bzl
+++ b/remote_emscripten_repository.bzl
@@ -62,6 +62,7 @@ def create_toolchains(name, repo_name, exec_compatible_with):
             "@emsdk//emscripten_toolchain:env.sh",
             "@emsdk//emscripten_toolchain:env.bat",
             "@nodejs//:node_files",
+            "@python_3_12_host//:py3_runtime",
         ],
     )
 
@@ -111,6 +112,7 @@ def create_toolchains(name, repo_name, exec_compatible_with):
         em_config = "@emscripten_cache//:emscripten_config",
         emscripten_binaries = repo_compiler_files_target,
         nodejs_bin = "@nodejs//:node",
+        python_bin = "@python_3_12_host//:py3_runtime",
         script_extension = select({
             "@bazel_tools//src/conditions:host_windows": "bat",
             "//conditions:default": "sh",
