diff -Naur grpc-original/BUILD grpc-modified/BUILD
--- grpc-original/BUILD	2026-01-18 10:36:57.927551089 +0000
+++ grpc-modified/BUILD	2026-01-18 10:38:13.700330753 +0000
@@ -34,7 +34,7 @@
     default_visibility = ["//visibility:public"],
     features = [
         "-parse_headers",
-        "layering_check",
+        "-layering_check",
     ],
 )
 
diff -Naur grpc-original/MODULE.bazel grpc-modified/MODULE.bazel
--- grpc-original/MODULE.bazel	2026-01-18 10:36:57.930551078 +0000
+++ grpc-modified/MODULE.bazel	2026-01-18 10:38:26.759306619 +0000
@@ -19,6 +19,7 @@
     repo_name = "com_github_grpc_grpc",
 )
 
+
 # Regular dependencies
 # ====================
 
@@ -68,7 +69,6 @@
     "3.11",
     "3.12",
     "3.13",
-    "3.14.0b2",
 ]
 
 python = use_extension("@rules_python//python/extensions:python.bzl", "python")
diff -Naur grpc-original/bazel/grpc_build_system.bzl grpc-modified/bazel/grpc_build_system.bzl
--- grpc-original/bazel/grpc_build_system.bzl	2026-01-18 10:36:57.931551075 +0000
+++ grpc-modified/bazel/grpc_build_system.bzl	2026-01-18 10:38:13.701330752 +0000
@@ -27,7 +27,6 @@
 Contains macros used throughout the repo.
 """
 
-load("@build_bazel_apple_support//rules:universal_binary.bzl", "universal_binary")
 load("@build_bazel_rules_apple//apple:ios.bzl", "ios_unit_test")
 load("@build_bazel_rules_apple//apple/testing/default_runner:ios_test_runner.bzl", "ios_test_runner")
 load("@com_google_protobuf//bazel:cc_proto_library.bzl", "cc_proto_library")
@@ -195,16 +194,11 @@
         srcs = srcs,
         deps = deps,
     )
-    universal_binary(
-        name = name + "_universal",
-        binary = name + "_native",
-    )
 
     # In order to avoid warnings from Bazel, names of the rule and its output file must differ.
     native.genrule(
         name = name,
         srcs = select({
-            "@platforms//os:macos": [name + "_universal"],
             "//conditions:default": [name + "_native"],
         }),
         outs = [name + "_binary"],
diff -Naur grpc-original/src/core/BUILD grpc-modified/src/core/BUILD
--- grpc-original/src/core/BUILD	2026-01-18 10:36:57.988550880 +0000
+++ grpc-modified/src/core/BUILD	2026-01-18 10:38:13.702330751 +0000
@@ -29,7 +29,7 @@
 package(
     default_visibility = ["//:__subpackages__"],
     features = [
-        "layering_check",
+        "-layering_check",
     ],
 )
 
diff -Naur grpc-original/src/core/lib/promise/detail/promise_factory.h grpc-modified/src/core/lib/promise/detail/promise_factory.h
--- grpc-original/src/core/lib/promise/detail/promise_factory.h	2026-01-18 10:36:58.071550596 +0000
+++ grpc-modified/src/core/lib/promise/detail/promise_factory.h	2026-01-18 10:38:13.702330751 +0000
@@ -125,17 +125,16 @@
 // Promote a callable(A) -> T | Poll<T> to a PromiseFactory(A) -> Promise<T> by
 // capturing A.
 template <typename Token, typename A, typename F>
-GPR_ATTRIBUTE_ALWAYS_INLINE_FUNCTION
-    absl::enable_if_t<!IsVoidCallable<ResultOf<F(A)>>::value,
-                      PromiseLike<Curried<RemoveCVRef<F>, A>>>
-    PromiseFactoryImpl(Token, F&& f, A&& arg) {
+absl::enable_if_t<!IsVoidCallable<ResultOf<F(A)>>::value,
+                  PromiseLike<Curried<RemoveCVRef<F>, A>>>
+PromiseFactoryImpl(Token, F&& f, A&& arg) {
   return Curried<RemoveCVRef<F>, A>(std::forward<F>(f), std::forward<A>(arg));
 }
 
 // Promote a callable() -> T|Poll<T> to a PromiseFactory(A) -> Promise<T>
 // by dropping the argument passed to the factory.
 template <typename A, typename F>
-GPR_ATTRIBUTE_ALWAYS_INLINE_FUNCTION inline absl::enable_if_t<
+inline absl::enable_if_t<
     !IsVoidCallable<ResultOf<F()>>::value, PromiseLike<RemoveCVRef<F>>>
 PromiseFactoryImpl(OnceToken, F f, A&&) {
   return PromiseLike<F>(std::move(f));
@@ -143,7 +142,7 @@
 
 // Promote a callable() -> Poll<T> to a PromiseFactory() -> Promise<T>
 template <typename F>
-GPR_ATTRIBUTE_ALWAYS_INLINE_FUNCTION inline absl::enable_if_t<
+inline absl::enable_if_t<
     !IsVoidCallable<ResultOf<F()>>::value &&
         PollTraits<ResultOf<F()>>::is_poll(),
     PromiseLike<RemoveCVRef<F>>>
@@ -153,7 +152,7 @@
 
 // Promote a callable() -> T to a PromiseFactory() -> Immediate<T>
 template <typename F>
-GPR_ATTRIBUTE_ALWAYS_INLINE_FUNCTION inline auto PromiseFactoryImpl(
+inline auto PromiseFactoryImpl(
     std::enable_if_t<!IsVoidCallable<ResultOf<F()>>::value &&
                          !PollTraits<ResultOf<F()>>::is_poll() &&
                          !std::is_same_v<ResultOf<F()>, void>,
@@ -163,7 +162,7 @@
   return PromiseLike<decltype(f2)>(std::move(f2));
 }
 template <typename F>
-GPR_ATTRIBUTE_ALWAYS_INLINE_FUNCTION inline auto PromiseFactoryImpl(
+inline auto PromiseFactoryImpl(
     std::enable_if_t<!IsVoidCallable<ResultOf<F()>>::value &&
                          !PollTraits<ResultOf<F()>>::is_poll() &&
                          std::is_same_v<ResultOf<F()>, void>,
@@ -176,7 +175,7 @@
 
 // Given a callable(A) -> Promise<T>, name it a PromiseFactory and use it.
 template <typename Token, typename A, typename F>
-GPR_ATTRIBUTE_ALWAYS_INLINE_FUNCTION inline absl::enable_if_t<
+inline absl::enable_if_t<
     IsVoidCallable<ResultOf<F(A)>>::value,
     PromiseLike<decltype(std::declval<F>()(std::declval<A>()))>>
 PromiseFactoryImpl(Token, F&& f, A&& arg) {
@@ -185,7 +184,7 @@
 
 // Given a callable(A) -> Promise<T>, name it a PromiseFactory and use it.
 template <typename Token, typename A, typename F>
-GPR_ATTRIBUTE_ALWAYS_INLINE_FUNCTION inline absl::enable_if_t<
+inline absl::enable_if_t<
     IsVoidCallable<ResultOf<F(A)>>::value,
     PromiseLike<decltype(std::declval<F>()(std::declval<A>()))>>
 PromiseFactoryImpl(Token, F& f, A&& arg) {
@@ -195,19 +194,17 @@
 // Given a callable() -> Promise<T>, promote it to a
 // PromiseFactory(A) -> Promise<T> by dropping the first argument.
 template <typename Token, typename A, typename F>
-GPR_ATTRIBUTE_ALWAYS_INLINE_FUNCTION
-    absl::enable_if_t<IsVoidCallable<ResultOf<F()>>::value,
-                      PromiseLike<decltype(std::declval<F>()())>>
-    PromiseFactoryImpl(Token, F&& f, A&&) {
+absl::enable_if_t<IsVoidCallable<ResultOf<F()>>::value,
+                  PromiseLike<decltype(std::declval<F>()())>>
+PromiseFactoryImpl(Token, F&& f, A&&) {
   return f();
 }
 
 // Given a callable() -> Promise<T>, name it a PromiseFactory and use it.
 template <typename Token, typename F>
-GPR_ATTRIBUTE_ALWAYS_INLINE_FUNCTION
-    absl::enable_if_t<IsVoidCallable<ResultOf<F()>>::value,
-                      PromiseLike<decltype(std::declval<F>()())>>
-    PromiseFactoryImpl(Token, F&& f) {
+absl::enable_if_t<IsVoidCallable<ResultOf<F()>>::value,
+                  PromiseLike<decltype(std::declval<F>()())>>
+PromiseFactoryImpl(Token, F&& f) {
   return f();
 }
 
@@ -225,10 +222,10 @@
                                               std::declval<A>()));
   using UnderlyingFactory = F;
 
-  GPR_ATTRIBUTE_ALWAYS_INLINE_FUNCTION explicit OncePromiseFactory(F f)
+  explicit OncePromiseFactory(F f)
       : f_(std::move(f)) {}
 
-  GPR_ATTRIBUTE_ALWAYS_INLINE_FUNCTION Promise Make(Arg&& a) {
+  Promise Make(Arg&& a) {
     return PromiseFactoryImpl(OnceToken{}, std::move(f_), std::forward<Arg>(a));
   }
 };
@@ -243,10 +240,10 @@
   using Promise = decltype(PromiseFactoryImpl(OnceToken{}, std::move(f_)));
   using UnderlyingFactory = F;
 
-  GPR_ATTRIBUTE_ALWAYS_INLINE_FUNCTION explicit OncePromiseFactory(F f)
+  explicit OncePromiseFactory(F f)
       : f_(std::move(f)) {}
 
-  GPR_ATTRIBUTE_ALWAYS_INLINE_FUNCTION Promise Make() {
+  Promise Make() {
     return PromiseFactoryImpl(OnceToken{}, std::move(f_));
   }
 };
@@ -265,13 +262,13 @@
   using Promise =
       decltype(PromiseFactoryImpl(RepeatableToken{}, f_, std::declval<A>()));
 
-  GPR_ATTRIBUTE_ALWAYS_INLINE_FUNCTION explicit RepeatedPromiseFactory(F f)
+  explicit RepeatedPromiseFactory(F f)
       : f_(std::move(f)) {}
 
-  GPR_ATTRIBUTE_ALWAYS_INLINE_FUNCTION Promise Make(Arg&& a) const {
+  Promise Make(Arg&& a) const {
     return PromiseFactoryImpl(RepeatableToken{}, f_, std::forward<Arg>(a));
   }
-  GPR_ATTRIBUTE_ALWAYS_INLINE_FUNCTION Promise Make(Arg&& a) {
+  Promise Make(Arg&& a) {
     return PromiseFactoryImpl(RepeatableToken{}, f_, std::forward<Arg>(a));
   }
   template <typename U>
@@ -293,13 +290,13 @@
   using Arg = void;
   using Promise = decltype(PromiseFactoryImpl(RepeatableToken{}, f_));
 
-  GPR_ATTRIBUTE_ALWAYS_INLINE_FUNCTION explicit RepeatedPromiseFactory(F f)
+  explicit RepeatedPromiseFactory(F f)
       : f_(std::move(f)) {}
 
-  GPR_ATTRIBUTE_ALWAYS_INLINE_FUNCTION Promise Make() const {
+  Promise Make() const {
     return PromiseFactoryImpl(RepeatableToken{}, f_);
   }
-  GPR_ATTRIBUTE_ALWAYS_INLINE_FUNCTION Promise Make() {
+  Promise Make() {
     return PromiseFactoryImpl(RepeatableToken{}, f_);
   }
 };
diff -Naur grpc-original/src/proto/grpc/health/v1/BUILD grpc-modified/src/proto/grpc/health/v1/BUILD
--- grpc-original/src/proto/grpc/health/v1/BUILD	2026-01-18 10:36:58.134550381 +0000
+++ grpc-modified/src/proto/grpc/health/v1/BUILD	2026-01-18 10:38:13.702330751 +0000
@@ -44,3 +44,10 @@
         "health.proto",
     ],
 )
+
+# This alias is required by Envoy's build
+alias(
+    name = "_health_proto_only",
+    actual = ":health_proto",
+    visibility = ["//visibility:public"],
+)
diff -Naur grpc-original/third_party/BUILD grpc-modified/third_party/BUILD
--- grpc-original/third_party/BUILD	2026-01-18 10:36:58.247549994 +0000
+++ grpc-modified/third_party/BUILD	2026-01-18 10:38:55.709243007 +0000
@@ -1,4 +1,4 @@
-load("@bazel_skylib//rules:common_settings.bzl", "bool_flag")
+load("@bazel_skylib//rules:common_settings.bzl", "bool_flag", "label_flag")
 
 exports_files([
     "gtest.BUILD",
@@ -31,21 +31,43 @@
     flag_values = {":grpc_use_openssl": "true"},
 )
 
+label_flag(
+    name = "ssl_lib",
+    build_setting_default = ":_default_ssl",
+    visibility = ["//visibility:public"],
+)
+
+label_flag(
+    name = "crypto_lib",
+    build_setting_default = ":_default_crypto",
+    visibility = ["//visibility:public"],
+)
+
 alias(
-    name = "libssl",
+    name = "_default_ssl",
     actual = select({
         ":grpc_use_openssl_setting": "@openssl//:ssl",
         "//conditions:default": "@boringssl//:ssl",
     }),
-    tags = ["manual"],
 )
 
 alias(
-    name = "libcrypto",
+    name = "_default_crypto",
     actual = select({
         ":grpc_use_openssl_setting": "@openssl//:crypto",
         "//conditions:default": "@boringssl//:crypto",
     }),
+)
+
+alias(
+    name = "libssl",
+    actual = ":ssl_lib",
+    tags = ["manual"],
+)
+
+alias(
+    name = "libcrypto",
+    actual = ":crypto_lib",
     tags = ["manual"],
 )
 
