--- a/toolchain/internal/configure.bzl	2026-08-10 09:43:24.417953922 +0000
+++ b/toolchain/internal/configure.bzl	2026-08-10 09:48:55.909327590 +0000
@@ -753,10 +753,10 @@
     ],
 )
 
-filegroup(name = "all-files-{suffix}", srcs = [":all-components-{suffix}", {extra_files_str}])
+filegroup(name = "all-files-{suffix}", srcs = [":all-components-{suffix}", "redacted_dates.h", {extra_files_str}])
 filegroup(name = "archiver-files-{suffix}", srcs = [{extra_files_str}])
 filegroup(name = "assembler-files-{suffix}", srcs = [{extra_files_str}])
-filegroup(name = "compiler-files-{suffix}", srcs = [":compiler-components-{suffix}", {extra_files_str}])
+filegroup(name = "compiler-files-{suffix}", srcs = [":compiler-components-{suffix}", "redacted_dates.h", {extra_files_str}])
 filegroup(name = "dwp-files-{suffix}", srcs = [{extra_files_str}])
 filegroup(name = "linker-files-{suffix}", srcs = [":linker-components-{suffix}", {extra_files_str}])
 filegroup(name = "objcopy-files-{suffix}", srcs = [{extra_files_str}])
@@ -807,10 +807,10 @@
     ],
 )
 
-filegroup(name = "all-files-{suffix}", srcs = [":all-components-{suffix}", {extra_files_str}])
+filegroup(name = "all-files-{suffix}", srcs = [":all-components-{suffix}", "redacted_dates.h", {extra_files_str}])
 filegroup(name = "archiver-files-{suffix}", srcs = ["{llvm_dist_label_prefix}ar", {extra_files_str}])
 filegroup(name = "assembler-files-{suffix}", srcs = ["{llvm_dist_label_prefix}as", {extra_files_str}])
-filegroup(name = "compiler-files-{suffix}", srcs = [":compiler-components-{suffix}", {extra_files_str}])
+filegroup(name = "compiler-files-{suffix}", srcs = [":compiler-components-{suffix}", "redacted_dates.h", {extra_files_str}])
 filegroup(name = "dwp-files-{suffix}", srcs = ["{llvm_dist_label_prefix}dwp", {extra_files_str}])
 filegroup(name = "linker-files-{suffix}", srcs = [":linker-components-{suffix}", {extra_files_str}])
 filegroup(name = "objcopy-files-{suffix}", srcs = ["{llvm_dist_label_prefix}objcopy", {extra_files_str}])
